lib/pbench/server/api/resources/api_key.py

```diff
@@ -64,23 +67,24 @@ def _get(
             Success: 200 with response containing the list of api_key
 
         Raises:
-            APIAbort, reporting "UNAUTHORIZED"
+            APIAbort, reporting "UNAUTHORIZED" and "NOT_FOUND"
         """
         user = Auth.token_auth.current_user()
+        key_id = params.uri.get("key")
 
         if not user:
             raise APIAbort(
                 HTTPStatus.UNAUTHORIZED,
                 "User provided access_token is invalid or expired",
             )
 
-        key = APIKey.query(user=user)
-        key_dict: JSONOBJECT = {}
-        if key:
-            for i in key:
-                key_dict[i.id] = {"name": i.name, "key": i.api_key}
-
-        response = {"api_key": key_dict}
+        if not key_id:
+            keys = APIKey.query(user=user)
+        else:
+            keys = APIKey.query(id=key_id, user=user)
+            if not keys:
+                raise APIAbort(HTTPStatus.NOT_FOUND, "Requested key not found")
```

Comment:
I don't see any problem here. If you ask for "all my keys" and there are none, that's fine; it's just an empty list, not an error. If you ask for "this key" and it doesn't exist, that's a `NOT_FOUND`.

I don't see any point in polymorphing the result: an array of a single entry seems fine for the "id" case.

Reply:
> an array of a single entry seems fine for the "id" case.

It's OK, but...given that I asked for "this key", why do I have to pick it out of a list (of one key)?

My point is, we have two arguably _separate_ interfaces, here -- `GET /key` and `GET /key/<id>` -- I'm good with the first returning a list (which might be empty), and I'm OK with the second returning `NOT_FOUND`, but I don't see why the second interface should return a list on success when we know that, at best, it will contain only one result.